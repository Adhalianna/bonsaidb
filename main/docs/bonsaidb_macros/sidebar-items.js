initSidebarItems({"derive":[["Api","Derives the `bonsaidb::core::api::Api` trait. `#[api(name = \"Name\", authority = \"Authority\", response = ResponseType, error = ErrorType, core = bonsaidb::core]` `authority`, `response`, `error` and `core` are optional"],["Collection","Derives the `bonsaidb::core::schema::Collection` trait. `#[collection(authority = \"Authority\", name = \"Name\", views = [a, b, c])]`"],["Key","Derives the `bonsaidb::core::key::Key` trait."],["Schema","Derives the `bonsaidb::core::schema::Schema` trait. `#[schema(name = \"Name\", authority = \"Authority\", collections = [A, B, C]), core = bonsaidb::core]` `authority`, `collections` and `core` are optional"],["View","Derives the `bonsaidb::core::schema::View` trait. `#[view(collection=CollectionType, key=KeyType, value=ValueType, name = \"by-name\")]` `name` and `value` are optional"]]});