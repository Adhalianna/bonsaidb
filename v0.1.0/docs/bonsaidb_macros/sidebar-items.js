initSidebarItems({"derive":[["Action","Derives the `bonsaidb_core::permissions::Action` trait."]]});