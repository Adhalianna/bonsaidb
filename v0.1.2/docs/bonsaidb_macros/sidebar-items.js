initSidebarItems({"derive":[["Collection","Derives the `bonsaidb::core::schema::Collection` trait. `#[collection(authority = \"Authority\", name = \"Name\", views = [a, b, c])]`"],["Schema","Derives the `bonsaidb::core::schema::Schema` trait. `#[schema(name = \"Name\", authority = \"Authority\", collections = [A, B, C]), core = bonsaidb::core]` `authority`, `collections` and `core` are optional"],["View","Derives the `bonsaidb::core::schema::View` trait. `#[view(collection=CollectionType, key=KeyType, value=ValueType, name = \"by-name\")]` `name` and `value` are optional"]]});