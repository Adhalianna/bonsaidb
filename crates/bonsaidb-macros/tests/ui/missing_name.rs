use bonsaidb_macros::Collection;

#[derive(Collection)]
#[collection(authority = "hello")]
struct Test;

fn main() {}
