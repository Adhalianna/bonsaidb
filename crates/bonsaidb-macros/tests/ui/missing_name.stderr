error: You need to specify the collection name via `#[collection(name="name")]`
 --> tests/ui/missing_name.rs:3:10
  |
3 | #[derive(Collection)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the derive macro `Collection` (in Nightly builds, run with -Z macro-backtrace for more info)
